infrastructure-formula/python/pillar/network.py

```diff
@@ -0,0 +1,161 @@
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+from logging import getLogger
+
+from yaml import safe_load
+
+root = '/srv/salt-git/pillar'
+
+log = getLogger(__name__).debug
+
+def generate_network_pillar(enabled_domains, domain, host):
+    pillar = {}
+
+    if domain not in enabled_domains:
+        return pillar
+
+    domaindir = f'{root}/domain/{domain.replace(".", "_")}'
+
+    msg = f'common.network, host {host}:'
+
+    domaindata = {
+            'hosts': {},
+            'networks': {},
+    }
+
+    for file in domaindata.keys():
+        with open(f'{domaindir}/{file}.yaml') as fh:
+            domaindata[file] = safe_load(fh)
+
+    if host in domaindata['hosts']:
+        hostconfig = domaindata['hosts'][host]
+        ifconfig   = hostconfig.get('interfaces', {})
+
+        if 'primary_interface' in hostconfig:
+            primary_interface = hostconfig['primary_interface']
+
+        elif len(ifconfig) == 1:
+            primary_interface = next(iter(ifconfig))
+
+        else:
+            interface_candidates = [interface for interface in ifconfig.keys() if not interface.endswith('-ur')]
+
+            if len(interface_candidates) == 1:
+                primary_interface = next(iter(interface_candidates))
+
+            else:
+                primary_interface = 'eth0'
+
+        log(f'{msg} primary interface set to {primary_interface}')
+
+        if primary_interface in ifconfig and ( 'ip4' in ifconfig[primary_interface] or 'ip6' in ifconfig[primary_interface] ):
+            ip4 = ifconfig[primary_interface].get('ip4')
+            ip6 = ifconfig[primary_interface].get('ip6')
+
+            # if an interface name contains a hyphen, it can be assumed to match our short VLAN nameing convention
+            if '-' in primary_interface:
+                shortnet = primary_interface
+
+            # alternatively assess the network segment based off the source (hypervisor) interface
+            else:
+                shortnet = ifconfig[primary_interface].get('source', '').replace('x-', '')
+
+            log(f'{msg} shortnet set to {shortnet}')
+
+        # if no primary interface is available, find and apply addresses defined outside of an "interfaces" block (single interface hosts might use this)
+        else:
+            log(f'{msg} trying to use generic interface addresses')
+            ip4 = hostconfig.get('ip4')
+            ip6 = hostconfig.get('ip6')
+            ifconfig = {}
+            shortnet = None
+
+        ifconfig.pop(primary_interface, None)
+
+        log(f'{msg} primary IPv4 address set to {ip4}')
+        log(f'{msg} primary IPv6 address set to {ip6}')
+
+    # machine not in hosts, can be an error or expected (for example because the machine is bare metal or in an unsupported location)
+    else:
+        log(f'{msg} no host entry!')
+        return pillar
```

Comment:
70 lines above, you define `pillar = {}`, and it wasn't changed since then - so effectively this line means `return {}`.

Would it make sense to write it as `return {}` to make that clear, or do you have any plans to add something to `pillar` in the lines above?

(Similar question for line 30 - but at least that's closer to the definition.)

Reply:
Alternative idea: make this

```
if host not in domaindata['hosts']:
    return {}  # or   return pillar
```
and move it above line 45.

Then un-indent everything that currently lives inside `if host in domaindata['hosts']:` ;-)

Reply:
It's indeed equivalent. If you want I can define pillar only later, and make this `{}` explicit.

The early return with less indentation is a good idea, thanks.

Reply:
I tend to prefer ;-) `return {}` to make clear what's going on.

Reply:
I think now the variable is clearly defined a few lines above

Reply:
Looks good - thanks!